sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt

```diff
@@ -57,15 +56,20 @@ internal object KlaviyoNetworkMonitor : NetworkMonitor {
     /**
      * Check what type of network connection is currently servicing the device
      *
-     * @return Integer representing the current network type
+     * @return The current network type
      */
-    override fun getNetworkType(): Int {
-        if (connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true) {
-            return NetworkMonitor.NetworkType.WIFI.position
-        } else if (connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)?.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) == true) {
-            return NetworkMonitor.NetworkType.CELL.position
+    override fun getNetworkType(): NetworkMonitor.NetworkType {
+        val net = connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+
+        return if (net?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true) {
+            NetworkMonitor.NetworkType.Wifi
+        } else if (net?.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) == true ||
```

Comment:
Technically internet can come from a few other transport methods like bluetooth, so this will just catch those and treat them as cellular (maybe we should call our types something more generic like Stable and Unstable idk)

Reply:
Neat!